packages/features/ee/dsync/lib/handleGroupEvents.ts

```diff
@@ -0,0 +1,178 @@
+import type { DirectorySyncEvent, Group } from "@boxyhq/saml-jackson";
+
+import jackson from "@calcom/features/ee/sso/lib/jackson";
+import { createAProfileForAnExistingUser } from "@calcom/lib/createAProfileForAnExistingUser";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma from "@calcom/prisma";
+import { teamMetadataSchema } from "@calcom/prisma/zod-utils";
+import {
+  getTeamOrThrow,
+  sendSignupToOrganizationEmail,
+  sendExistingUserTeamInviteEmails,
+} from "@calcom/trpc/server/routers/viewer/teams/inviteMember/utils";
+
+import createUsersAndConnectToOrg from "./users/createUsersAndConnectToOrg";
+
+const handleGroupEvents = async (event: DirectorySyncEvent, orgId: number) => {
+  const { dsyncController } = await jackson();
+  // Find the group name associated with the event
+  const eventData = event.data as Group;
+
+  // If the group doesn't have any members assigned then return early
+  if (!eventData.raw.members.length) {
+    return;
+  }
+
+  const groupNames = await prisma.dSyncTeamGroupMapping.findMany({
+    where: {
+      directoryId: event.directory_id,
+      groupName: eventData.name,
+      orgId,
+    },
+    select: {
+      teamId: true,
+      team: {
+        include: {
+          parent: {
+            include: {
+              organizationSettings: true,
+            },
+          },
+          organizationSettings: true,
+        },
+      },
+      groupName: true,
+    },
+  });
+
+  if (!groupNames.length) {
+    return;
+  }
+
+  const org = await getTeamOrThrow(orgId);
+
+  // Check if the group member display property is an email
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const isEmail = emailRegex.test(eventData.raw.members[0].display);
+
+  let userEmails: string[] = [];
+
+  // TODO: Handle the case where display property is not an email
```

Comment:
Reached out to Boxyhq to see how to handle this. It would be nice if we could just make one call to grab all user data in a group. If not then we will have to make calls for each individual user.